### ISSUE TITLE
FacilityInfo: delete getRemoteJobManager and rearrange RemoteJob related stuff

### ISSUE THREAD

**Issue by mantid-builder** [mantidproject/mantid#12212]
This issue was originally [TRAC 11373](http://trac.mantidproject.org/mantid/ticket/11373)

Original Reporter: @FedeMPouzols 

This ticket is blocked by : 
- [X] [TRAC11124](http://trac.mantidproject.org/mantid/ticket/11124)
- [X] [TRAC11126](http://trac.mantidproject.org/mantid/ticket/11126)

`getRemoteJobManager()` is superseded by the RemoteJobManagerFactory (http://trac.mantidproject.org/mantid/ticket/11124).

When the remote algorithms are reorganized (http://trac.mantidproject.org/mantid/ticket/11126), FacilityInfo::getRemoteJobManager should not be needed any longer, and in general (I)RemoteJobManager should not be used in FacilityInfo.

Note: in http://trac.mantidproject.org/mantid/ticket/11126, rather than revamping the v1 remote algorithms breaking backwards compatibility, new version 2 remote algorithms have been introduced. Versions 1 still use `getRemoteJobManager()` and related chunks of code in `FacilityInfo`. So this ticket has to wait until the version 1 remove algorithms are removed (for which we could create a new ticket, or we could do it as part of this ticket).

**Comment by FedeMPouzols** [comment]
This can only be done when/if it is safe to remove version 1 of the remote algorithms, which is being used in some SNS reduction scripts.
